linkerd/access-log/src/tower.rs

```diff
@@ -0,0 +1,169 @@
+use futures::TryFuture;
+use pin_project::pin_project;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::time::{Duration, Instant};
+use tracing::{field, span, Level, Span};
+
+/// A tower layer that associates a tokio-tracing Span with each request
+#[derive(Clone)]
+pub struct AccessLogLayer {}
+
+#[derive(Clone)]
+pub struct AccessLogContext<Svc> {
+    inner: Svc,
```

Comment:
nit/style: generally we would just call this
```suggestion
pub struct AccessLogContext<S> {
    inner: S,
```